### ISSUE TITLE
[BITV] 9.4.1.2/2.3 - The "Open settings menu" menu - The aria-expanded attribute was correctly applied to the menu button, but the attribute always has the value "true" which is incorrect. (9)

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/server#37095]
Also, the use of the "nav" element only for the list of menu items is not correct. The "nav" element should also contain the menu button itself. See https://inclusive-components.design/menus-menu-buttons/ for more information.

Additionally, it is recommended not to use the status (open/close) of the menu as part of the accessible name. This status is communicated through the aria-expanded attribute.

The user's online status ("online", "do not disturb" etc.) should be reflected in the accessible name (e. g. instead of "open settings menu" the name should be something like "settings menu - BB - do not disturb" or "BB, do not disturb - settings").

- [x] https://github.com/nextcloud-libraries/nextcloud-vue/pull/4481
- [x] https://github.com/nextcloud-libraries/nextcloud-vue/pull/4487
- [x] Release
- [x] https://github.com/nextcloud/server/pull/40692
- [x] https://github.com/nextcloud/server/pull/40168

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v2-n3

</details>

- [x] 9.4.1.2/3.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v3-n3
- [x] 9.4.1.2/4.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v4-n3
- [x] 9.4.1.2/5.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v5-n3
- [x] 9.4.1.2/6.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v6-n3
- [x] 9.4.1.2/7.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v7-n3
- [x] 9.4.1.2/8.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v8-n3
- [x] 9.4.1.2/9.4 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v9-n4
- [x] 9.4.1.2/10.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v10-n3

**Comment by Pytal** [comment]
The implementation from https://inclusive-components.design/menus-menu-buttons/ suggests markup like
![image](https://github.com/nextcloud/server/assets/24800714/22e46bbd-3595-415c-af82-7041fdb33ed7)
explicitly without menu semantics due to:
![image](https://github.com/nextcloud/server/assets/24800714/d4c2de52-2325-4612-bbf1-3ca788e98825)

while W3C (https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/#sc1_label) does suggest using menu semantics
![image](https://github.com/nextcloud/server/assets/24800714/391f4cd2-2c39-4b2d-8511-1da9be642778)

How shall we proceed and determine which approach to use @michaelnissenbaum ?

**Comment by michaelnissenbaum** [comment]
The "Open settings menu" refers to a FlyOut menu, and you can find guidance on how to implement it at https://www.w3.org/WAI/tutorials/menus/flyout/. ARIA roles "menu" and "menuitem" are typically used for application menus, which we do not have in this context. You can refer to https://www.w3.org/TR/wai-aria-1.2/#menu, which states, "The menu role is appropriate when a list of menu items is presented in a manner similar to a menu on a desktop application." Additionally, it's important to note that in application menus, navigation is typically done using arrow keys rather than the TAB key.

**Comment by Pytal** [comment]
Thanks @michaelnissenbaum 

Would you consider the global search, notifications, and contacts search menus application menus?
![image](https://github.com/nextcloud/server/assets/24800714/baa08928-553d-40ed-8966-717f57aadf1a)

**Comment by michaelnissenbaum** [comment]
All three elements are not application menus. All three should be implemented as buttons (not as links) with the "aria-expanded" attribute.
